src/main/java/seedu/ddd/model/util/SampleDataUtil.java

```diff
@@ -6,46 +6,90 @@
-    private static final Contact[] sampleContacts = new Contact[] {
-        sampleCLient, sampleVendor
-    };
-    private static final int sampleNextContactId = sampleContacts.length;
-    private static final int sampleNextEventId = 1;
+
+    // Client fields
+    public static final String SAMPLE_CLIENT_NAME = "Amy Bee";
+    public static final String SAMPLE_CLIENT_PHONE = "12345678";
+    public static final String SAMPLE_CLIENT_EMAIL = "amy@example.com";
+    public static final String SAMPLE_CLIENT_ADDRESS = "Block 312, Amy Street 1";
+    public static final String SAMPLE_CLIENT_ID = "0";
+
+    // Vendor fields
+    public static final String SAMPLE_VENDOR_NAME = "Bob Choo";
+    public static final String SAMPLE_VENDOR_PHONE = "87654321";
+    public static final String SAMPLE_VENDOR_EMAIL = "bob@example.com";
+    public static final String SAMPLE_VENDOR_ADDRESS = "Block 123, Bobby Street 3";
+    public static final String SAMPLE_VENDOR_SERVICE = "Catering";
+    public static final String SAMPLE_VENDOR_ID = "1";
+
+    // Common contact fields
+    public static final String SAMPLE_TAG_1 = "budget";
+    public static final String SAMPLE_TAG_2 = "vegetarian";
+
+    // Event fields
+    public static final String SAMPLE_EVENT_NAME = "Sample Wedding";
+    public static final String SAMPLE_EVENT_DESCRIPTION = "Wedding reception";
+    public static final String SAMPLE_EVENT_DATE = "2000-01-01";
+    public static final String SAMPLE_EVENT_ID = "0";
+
+    private static final int SAMPLE_NEXT_CONTACT_ID = 2;
+    private static final int SAMPLE_NEXT_EVENT_ID = 1;
 
     public static ReadOnlyAddressBook getSampleAddressBook() {
+        Client sampleClient = new Client(
+            new Name(SAMPLE_CLIENT_NAME),
+            new Phone(SAMPLE_CLIENT_PHONE),
+            new Email(SAMPLE_CLIENT_EMAIL),
+            new Address(SAMPLE_CLIENT_ADDRESS),
+            getTagSet(SAMPLE_TAG_1, SAMPLE_TAG_2),
+            new ContactId(SAMPLE_CLIENT_ID)
+        );
+        Vendor sampleVendor = new Vendor(
+            new Name(SAMPLE_VENDOR_NAME),
+            new Phone(SAMPLE_VENDOR_PHONE),
+            new Email(SAMPLE_VENDOR_EMAIL),
+            new Address(SAMPLE_VENDOR_ADDRESS),
+            new Service(SAMPLE_VENDOR_SERVICE),
+            getTagSet(SAMPLE_TAG_1),
+            new ContactId(SAMPLE_VENDOR_ID)
+        );
+        Event sampleEvent = new Event(
+            new Name(SAMPLE_EVENT_NAME),
+            new Description(SAMPLE_EVENT_DESCRIPTION),
+            new Date(SAMPLE_EVENT_DATE),
+            new EventId(SAMPLE_EVENT_ID)
+        );
+
+        sampleEvent.addClient(sampleClient);
+        sampleEvent.addVendor(sampleVendor);
+        sampleClient.addEvent(sampleEvent);
+        sampleVendor.addEvent(sampleEvent);
+
         AddressBook sampleAb = new AddressBook();
-        for (Contact sampleContact : sampleContacts) {
```

Comment:
Does this lead to the sample data always having at most 2 sampleContacts? 
Be careful that when adding sample data, we need to do sampleAb.addContact instead of just adding a new Contact. If extensibility is one of our goals, this might violate that.